authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/AuthorizationProperties.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization.ranger;
+
+import java.util.Map;
+
+public interface AuthorizationProperties {
+  /** The `Prefix.Wildcard` properties segment name */
+  void validateProperties(Map<String, String> properties);
+
+  static void validate(
+      Class<? extends AuthorizationProperties> authPropertiesClass,
+      Map<String, String> properties) {
+    try {
+      AuthorizationProperties authorizationProperties =
+          authPropertiesClass.getDeclaredConstructor().newInstance();
+      authorizationProperties.validateProperties(properties);
+    } catch (ReflectiveOperationException e) {
+      throw new IllegalArgumentException("Failed to instantiate AuthorizationProperties class", e);
+    }
```

Comment:
It seems unnecessary to use reflection in this method. 

Why don't we directly use `static void validate(AuthorizationProperties authorizationProperties, Map<String, String> properties)`?

Reply:
I am refactoring this change.